src/components/NewTodoForm/NewTodoForm.tsx

```diff
@@ -0,0 +1,100 @@
+import { useState } from 'react';
+import { User } from '../../types/User';
+import { NewTodo } from '../../types/NewTodo';
+
+interface Props {
+  addTodo: (newTodo: NewTodo) => void;
+  users: User[];
+}
+
+export const CreateNewTodo = ({ addTodo, users }: Props) => {
+  const [title, setTitle] = useState('');
+  const [isTitleHasError, setisTitleHasError] = useState(false);
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [isSelectHasError, setSelectHasError] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setisTitleHasError(false);
+  };
+
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUserId(+event.target.value);
+    setSelectHasError(false);
+  };
+
+  const handleFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const trimmedTitle = !title.trim();
+
+    if (trimmedTitle && !selectedUserId) {
+      setisTitleHasError(true);
+      setSelectHasError(true);
+
+      return;
+    }
+
+    if (trimmedTitle) {
+      setisTitleHasError(true);
+
+      return;
+    }
+
+    if (!selectedUserId) {
```

Comment:
All if-blocks can be placed in a separate function. This will make the code more organized and readable